### ISSUE TITLE
[0.9.5] metaqueries via curl hanging, succeeding via CLI

### ISSUE THREAD

**Issue by rackunits** [influxdata/influxdb#4929]
When dropping a database on RC2 9.5.0 I don't get a response.

https://www.dropbox.com/s/fyezv0z6kupge9x/Screenshot%202015-11-30%2013.32.16.png?dl=0

**Comment by beckettsean** [comment]
That is intended behavior. `CREATE DATABASE` also returns nothing on success.

```
$ curl -Gi http://localhost:8086/query --data-urlencode 'q=create database foo'
HTTP/1.1 200 OK
Content-Type: application/json
Request-Id: 556b6a77-97c0-11e5-93ee-000000000000
X-Influxdb-Version: 0.9.5
Date: Tue, 01 Dec 2015 00:13:22 GMT
Content-Length: 16

{"results":[{}]}

$ curl -Gi http://localhost:8086/query --data-urlencode 'q=show databases'
HTTP/1.1 200 OK
Content-Type: application/json
Request-Id: 5c28beb0-97c0-11e5-93f0-000000000000
X-Influxdb-Version: 0.9.5
Date: Tue, 01 Dec 2015 00:13:34 GMT
Content-Length: 121

{"results":[{"series":[{"name":"databases","columns":["name"],"values":[["mydb"],["_internal"],["telegraf"],["foo"]]}]}]}

$ curl -Gi http://localhost:8086/query --data-urlencode 'q=drop database foo'
HTTP/1.1 200 OK
Content-Type: application/json
Request-Id: 6208d764-97c0-11e5-93f1-000000000000
X-Influxdb-Version: 0.9.5
Date: Tue, 01 Dec 2015 00:13:43 GMT
Content-Length: 16

{"results":[{}]}
```

**Comment by rackunits** [comment]
I see, however, it does not return the empty result set, It returns nothing, the curl process hangs unless I add a timeout. The log shows the CREATE DATABASE but the curl process does not receive the response.

This sometimes also happens with select statements, that work fluently in the shell but not when curling the machine.

**Comment by beckettsean** [comment]
@rackunits thanks for the clarification, I must admit I was reading quickly and didn't see the timeouts in your curl image. In the future please copy/paste text, which is easier to parse.

> This sometimes also happens with select statements, that work fluently in the shell but not when curling the machine.

That's very confusing, because the CLI uses the same API and endpoints as `curl` statements. Can you give an example of a query that hangs in `curl` but succeeds in the CLI?

Can you share the logs from those incidents?

**Comment by rackunits** [comment]
Dear,

It took me a while to reproduce this issue.
After a while I gave up and continued developing, then all of a sudden I had the same issue with a DROP DATABASE statement.

Evidence: https://www.dropbox.com/s/3o4v6rv24snygxz/drop-database-no-response2.gif?dl=0

I have made a snapshot of the machine to provide you with logs if needed.

Here is the log up until the time of this event: https://www.dropbox.com/s/hgn1h8y17cfc6am/influxd.log?dl=0

And the influxdb config I am using: https://www.dropbox.com/s/01fvypccl3fe1c6/influxdb.conf?dl=0

**Comment by rackunits** [comment]
Some additional info: when this occurs, later imports also fail.

curl -i -s -XPOST 'http://localhost:8086/write?db=metrics_1449055108' --data-binary @/home/330/_post_calc_/20151202111828/annotations.influx
Received status 500

The influxdb log:
[http] 2015/12/02 12:18:28 ::1 - - [02/Dec/2015:12:18:28 +0100] GET /query?pretty=true&q=DROP%20DATABASE%20databases HTTP/1.1 400 95 - curl/7.35.0 69a1c1b2-98e6-11e5-8248-000000000000 132.374µs
[query] 2015/12/02 12:18:29 DROP DATABASE name
[http] 2015/12/02 12:18:29 ::1 - - [02/Dec/2015:12:18:29 +0100] GET /query?pretty=true&q=DROP%20DATABASE%20name HTTP/1.1 200 94 - curl/7.35.0 6a3db831-98e6-11e5-8249-000000000000 205.113µs
[query] 2015/12/02 12:18:30 CREATE DATABASE metrics_1449055108
[metastore] 2015/12/02 12:18:30 database 'metrics_1449055108' created
[metastore] 2015/12/02 12:18:30 retention policy 'default' for database 'metrics_1449055108' created
[http] 2015/12/02 12:18:30 ::1 - - [02/Dec/2015:12:18:30 +0100] GET /query?q=CREATE%20DATABASE%20metrics_1449055108 HTTP/1.1 200 16 - curl/7.35.0 6ad8d3a3-98e6-11e5-824a-000000000000 7.12641ms
[http] 2015/12/02 12:18:35 ::1 - - [02/Dec/2015:12:18:30 +0100] POST /write?db=metrics_1449055108 HTTP/1.1 500 8 - curl/7.35.0 6adc7aad-98e6-11e5-824b-000000000000 5.003907851s

**Comment by beckettsean** [comment]
@rackunits when the `DROP DATABASE` command failed from `curl` did it work from the CLI? I see nothing in the GIF or the logs to suggest that it did.

A non-responsive database issuing 500s on writes and timeouts on all queries sounds like the deadlocking issue that was fixed in 0.9.5.1. However, if you can complete queries from the CLI and not with `curl` that is a different thing. However, there's no evidence yet of that happening. There's lots of evidence to show the `curl` queries hanging, but nothing showing that they work from the CLI. If the system is restarted the deadlock clears and all queries begin working again. Did you restart before trying the CLI?

**Comment by rackunits** [comment]
I apologize for the ambiguity in my response.
Because this issue was blocking my development, I set up an additional vm with InfuxDb 0.9.5.1.
The gif, logs and config are from the 0.9.5.1 release.
The point that I was trying to make is that the query can (select) or is being processed (drop), but no response is given.
The DROP actually took place, because the `SHOW DATABASES` at the end showed no databases, whilst the `curl` process was still hanging. That is why it was useless to issue the DROP command again from the cli (the db was already removed).

Please keep this issue open, while I will try to reproduce the issues I have (Drop and select statements not getting results from curl command). I think the first has been proven by my previous post.

I did not try to restart the system yet, because it takes a while for these issues to pop up.

**Comment by rackunits** [comment]
I think I narrowed it down a bit.

First of all something I want to clear out.

My initial statement was that the `curl` process was hanging while the CLI did return a response.
This was and is still true, but the statement was phrased this way because that was how I tested the issue. However I discovered that curling from the ubuntu shell while the `curl` process is hanging does return a response: https://www.dropbox.com/s/2yaapfd9qqno1rn/cul-select-hang.gif?dl=0
So it was just a simple query hang.

I added a timeout to the curl command and noticed that after the first timeout (10s in my case) all queries returned a result. Then I added a sleep right before my curl select commands and all of them passed: https://www.dropbox.com/s/ovf50a85d66t26m/cul-select-no-hang.gif?dl=0

I think the issue comes down to how fast you query the database after dumping data.
This is however only in case of the select statements, and this does not explain the behaviour for the drop database query.

**Comment by beckettsean** [comment]
@rackunits I have seen no other instances of `curl` hanging for users, so I can only suspect there is something local to your machine or your instance that is causing some issue. Without a reproducible case I cannot investigate any further.

Also, I don't understand that first GIF in your most recent comment. I see logs tailing, and a curl command with a query that returns immediately. I don't see any hanging curl statements, can you clarify and perhaps describe the picture a bit more? It might seem obvious to you what your environment is and what each panel is doing, but there's every possibility I'm misinterpreting the picture through ignorance.

Also, there's no way to pause and investigate, and the fonts are very very small. I realize you are trying to show evolution over time, but please give actual text in the issue and then describe the transitions.

**Comment by jsternberg** [comment]
I'm closing this because it is for an older version of InfluxDB (before we made a lot of fixes to just about everything). Please try the latest version available (0.13) and comment here if you are still experiencing this issue. Thank you.
